File: src/frontend/src/data/entities/project/ProjectVolatileState.ts
import { type ConnectorInstanceID } from "@common/data/entities/connector/ConnectorInstance.ts";
import { type ProjectID } from "@common/data/entities/project/Project.ts";
import { ProjectExternalState } from "@common/data/entities/project/ProjectExternalState.ts";

/**
 * Holds the external state for a specific project and connector instance.
 */
export class ProjectVolatileState {
    public readonly projectID: ProjectID;
    public readonly connectorInstance: ConnectorInstanceID;

    public readonly externalState: ProjectExternalState;

    public constructor(projectID: ProjectID, connectorInstance: ConnectorInstanceID, externalState: ProjectExternalState) {
        this.projectID = projectID;
        this.connectorInstance = connectorInstance;

        this.externalState = externalState;
    }
}

File: src/server/data/entities/project/project_volatile_state.py
from dataclasses import dataclass

from common.py.data.entities.connector import ConnectorInstanceID
from common.py.data.entities.project import ProjectExternalState, ProjectID


@dataclass(kw_only=True)
class ProjectVolatileState:
    """
    The volatile state of a project.

    Attributes:
        project_id: The project ID.
        connector_instance: The connector instaqnce ID.
        external_state: The project's external state.
        timestamp: The timestamp of the volatile state.
    """

    project_id: ProjectID
    connector_instance: ConnectorInstanceID

    external_state: ProjectExternalState

    timestamp: float = 0.0
